apps/talk_signaling.sh

```diff
@@ -0,0 +1,65 @@
+#!/bin/bash
+
+# T&M Hansson IT AB © - 2020, https://www.hanssonit.se/
+
+# shellcheck disable=2034,2059
+true
+# shellcheck source=lib.sh
+. <(curl -sL https://raw.githubusercontent.com/nextcloud/vm/master/lib.sh)
+
+# Check for errors + debug code and abort if something isn't right
+# 1 = ON
+# 0 = OFF
+DEBUG=0
+debug_mode
+
+DESCRIPTION="High Performance Backend for Talk"
+
+# Check if root
+root_check
+
+# Check if HPB is already installed
+is_process_running dpkg
+is_process_running apt
+print_text_in_color "$ICyan" "Checking if ${DESCRIPTION} is already installed..."
+if ! is_this_installed nextcloud-spreed-signaling
+then
+    choice=$(whiptail --radiolist "It seems like '${DESCRIPTION}' is already installed.\nChoose what you want to do.\nSelect by pressing the spacebar and ENTER" "$WT_HEIGHT" "$WT_WIDTH" 4 \
+    "Uninstall ${DESCRIPTION}" "" OFF \
+    "Reinstall ${DESCRIPTION}" "" ON 3>&1 1>&2 2>&3)
+    case "$choice" in
+        "Uninstall ${DESCRIPTION}")
+          # TODO: remove nats, janus and signaling-server
+          :
+        ;;
+        "Reinstall ${DESCRIPTION}")
+          # TODO: remove nats, janus and signaling-server
+          :
+        ;;
+        *)
+        ;;
+    esac
+else
+    print_text_in_color "$ICyan" "Installing ${DESCRIPTION}..."
+fi
+
+# Install
+. /etc/lsb-release
+for package in nats-server nextcloud-spreed-signaling janus
+do
+  curl -sL -o "/etc/apt/trusted.gpg.d/morph027-${key}.asc" "https://packaging.gitlab.io/${key}/gpg.key"
```

Comment:
We have a function called `curl_to_dir` which I think would fit here?

https://github.com/nextcloud/vm/blob/master/lib.sh#L306

Reply:
Sure, will add.

Reply:
:+1: